test/functorch/test_aotdispatch.py

```diff
@@ -2735,6 +2735,148 @@ def forward(self, *args):
 
 class TestAOTExport(AOTTestCase):
 
+    def test_aot_export_predispatch_func_simple(self):
+        def fn(p, x):
+            y = x + 2
+            with torch.no_grad():
+                y.add_(2)
+            return (x * 2 + y,)
+
+        mod = TestMod(fn)
+        inp = torch.randn(2, 2)
+
+        gm, _ = aot_export_module(mod, [inp], trace_joint=False, pre_dispatch=True)
+        self.assertExpectedInline(str(gm.code).strip(), """\
+def forward(self, arg0_1, arg1_1):
+    add = torch.ops.aten.add.Tensor(arg1_1, 2)
+    _set_grad_enabled = torch._C._set_grad_enabled(False)
+    add_1 = torch.ops.aten.add.Tensor(add, 2);  add = None
+    _set_grad_enabled_1 = torch._C._set_grad_enabled(False)
+    mul = torch.ops.aten.mul.Tensor(arg1_1, 2);  arg1_1 = None
+    add_2 = torch.ops.aten.add.Tensor(mul, add_1);  mul = add_1 = None
+    return (add_2,)""")
+
+    def test_aot_export_predispatch_func_composite_implicit(self):
```

Comment:
I guess this test explicitly does not do as instructed until the next PR in the stack 😛

Reply:
Hahah yeah maybe i should rename to func_still_decompose_compisite_implicit